cumulusci/tasks/datadictionary.py

```diff
@@ -101,72 +113,56 @@ def _init_options(self, kwargs):
             True if include_dependencies is None else include_dependencies
         )
 
-        if "additional_dependencies" in self.options and not all(
-            ["github" in dep for dep in self.options["additional_dependencies"]]
-        ):
-            raise TaskOptionsError("Only GitHub dependencies are currently supported.")
+        if "additional_dependencies" in self.options:
+            additional_deps = parse_dependencies(
+                self.options["additional_dependencies"]
+            )
+            if not all(isinstance(d, GitHubDynamicDependency) for d in additional_deps):
+                raise TaskOptionsError(
+                    "Only GitHub dependencies are currently supported."
+                )
 
     def _get_repo_dependencies(
-        self, dependencies=None, include_beta=None, visited_repos=None
-    ):
+        self, dependencies: List[GitHubDynamicDependency]
+    ) -> List[Package]:
         """Return a list of Package objects representing all of the GitHub repositories
         in this project's dependency tree. Ignore all non-GitHub dependencies."""
-        deps = []
-        visited = visited_repos or set()
-
-        for dependency in dependencies:
-            if "github" in dependency:
-                repo = self.project_config.get_repo_from_url(dependency["github"])
-                if repo is None:
-                    raise DependencyResolutionError(
-                        f"Github repository {dependency['github']} not found or not authorized."
-                    )
-                _, ref = self.project_config.get_ref_for_dependency(
-                    repo, dependency, include_beta
-                )
+        github_deps = set()
```

Comment:
Unlike dicts, sets are not guaranteed to return entries in insertion order. Does that matter here?

Reply:
I don't think so. The data dictionary generator makes no commitment about the ordering of output data, and if it did I think alphabetical by package name is more likely to be correct than dependency resolution order.